lib/krb5/fcache.c

```diff
@@ -1186,38 +1419,44 @@ fcc_get_cache_next(krb5_context context, krb5_cc_cursor cursor, krb5_ccache *id)
 {
     struct fcache_iter *iter = cursor;
     krb5_error_code ret;
-    const char *fn, *cc_type;
-    krb5_ccache cc;
+    char *name = NULL;
 
+    *id = NULL;
     if (iter == NULL)
         return krb5_einval(context, 2);
 
-    if (!iter->first) {
-	krb5_clear_error_message(context);
-	return KRB5_CC_END;
+    if (!iter->curr_location) {
+        /* Next name */
+        if ((ret = next_location(context, iter)))
+            return ret;
+        /* Output the current base location */
+        if ((ret = try1(context, iter->curr_location, id)) || *id)
+            return ret;
     }
-    iter->first = 0;
 
-    /*
-     * Note: do not allow krb5_cc_default_name() to recurse via
-     * krb5_cc_cache_match().
-     * Note that context->default_cc_name will be NULL even though
-     * KRB5CCNAME is set in the environment if
-     * krb5_cc_set_default_name() hasn't
-     */
-    fn = krb5_cc_default_name(context);
-    ret = krb5_cc_resolve(context, fn, &cc);
-    if (ret != 0)
-        return ret;
-    cc_type = krb5_cc_get_type(context, cc);
-    if (strcmp(cc_type, "FILE") != 0) {
-        krb5_cc_close(context, cc);
-        return KRB5_CC_END;
+    /* Look for subsidiaries of iter->curr_location */
+    if (!iter->d) {
+        free(iter->dname);
+        if ((iter->dname = my_dirname(iter->curr_location)) == NULL)
+            return krb5_enomem(context);
+        if ((iter->d = opendir(iter->dname)) == NULL) {
+            if ((ret = next_location(context, iter)))
+                return ret;
+            return fcc_get_cache_next(context, cursor, id);
+        }
     }
-
-    *id = cc;
-
-    return 0;
+    for (ret = next_dir_match(context, iter, &name);
+         ret == 0 && name != NULL;
+         ret = next_dir_match(context, iter, &name)) {
+        if ((ret = try1(context, name, id)) || *id) {
```

Comment:
Tabs here?

Reply:
No, that's 5-space indent for the `for` loop bits, and 4-space for the body.